### PR TITLE
@OR-1196 : Smart answers

### PR DESCRIPTION
**Pull request by mangogrin**

### CODE REVIEW

**Comment by kiwikid747** [comment]
👍

File: app/views/go-to-activities.mustache [hunk 1]
```diff
@@ -18,7 +20,11 @@
     </ol>
 
     <p class="form-group">
-      <a class="button" href="{{basePath}}/{{accountId}}/groups/{{group}}/activities/details" data-test="go-to-my-activities">{{#__}}go-to-activities.button-label{{/__}}</a>
+      <a class="button" href="{{basePath}}/{{accountId}}/groups/{{group}}/activities/details"
+         data-test="go-to-my-activities">{{#__}}go-to-activities.button-label{{/__}}</a>
     </p>
-  {{/applicationContent}}
+  </div>
+{{<smartAnswers}}{{/smartAnswers}}
```
Comment (kiwikid747) [inline app/views/go-to-activities.mustache]: This can be an inline import: {{>smartAnswers}}

File: app/view-models/smart-answers-view-model.js [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+        ({
+          categoryTitle: category.title,
+          categoryName: category.name,
+          sortedActivities: groupedActivities[category.name]
+            .map(activityDecorator)
+            .map(activity => ({
+              activityTitle: activity.title, activityName: activity.activity,
+            })),
+        }));
+    this.hidden = this.categoryGroups.length > 0 ? '' : 'hidden';
```
Comment (will518) [inline app/view-models/smart-answers-view-model.js]: I know we've done something similar elsewhere but does this couple the css classes too closely to the javascript? Would it be better to make this a boolean and use a mustache if clause e.g.  `{{#hidden}}hidden{{/hidden}}`?

Curious to get your thoughts @chilliwilly77 @kiwikid747.

File: app/controllers/activities-controller.js [hunk 3]
```diff
@@ -2,20 +2,27 @@ const express = require('express');
...
-  ActivitiesModel.findUnsortedByVersionAccountIdAndGroup(version, accountId, group)
-    .then(activities => {
-      res.render('activities', Object.assign({
-        accountId,
-        group,
-      }, new ActivitiesViewModel(activities, false, '', version, group)));
-    });
+  Promise.all([
+    ActivitiesModel.findUnsortedByVersionAccountIdAndGroup(version, accountId, group),
+    ActivitiesModel.findSortedByAccountIdAndGroupByCategory(accountId, version, group),
```
[the remark this answers is not in the document]
Comment (mangogrin) [reply app/controllers/activities-controller.js]: I will change the ordering of params to be consistent.

No, they differ with the returned model as well. One is grouped by category.

Even if they had the same functionality except sorted/unsorted i would keep it separate for more clarity.

File: app/controllers/activities-controller.js [hunk 4]
```diff
@@ -2,20 +2,27 @@ const express = require('express');
...
+  ]).then(results => {
```
[the remark this answers is not in the document]
Comment (mangogrin) [reply app/controllers/activities-controller.js]: `.then(([unsortedActivities, sortedActivities]) => {` 
works

File: app/views/go-to-activities.mustache (hunk 1 above)
Comment (mangogrin) [reply app/views/go-to-activities.mustache]: done

File: app/controllers/activities-controller.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (mangogrin) [reply app/controllers/activities-controller.js]: done

File: app/controllers/activities-controller.js (hunk 4 above)
[the remark this answers is not in the document]
Comment (mangogrin) [reply app/controllers/activities-controller.js]: done

File: app/controllers/categorise-activity-controller.js [hunk 5]
```diff
@@ -28,6 +30,16 @@ router.get('', validator.get, (req, res) => {
     { accountId, activityId, group, title, groupTitle }, categoryView));
 });
 
+function renderGotoActivitiesPage(accountId, version, group, res) {
+  ActivitiesModel.findSortedByAccountIdAndGroupByCategory(accountId, version, group)
+    .then((sortedActivities) => {
```
[the remark this answers is not in the document]
Comment (mangogrin) [reply app/controllers/categorise-activity-controller.js]: removed

File: app/view-models/smart-answers-view-model.js (hunk 2 above)
Comment (mangogrin) [reply app/view-models/smart-answers-view-model.js]: done

**Comment by mangogrin** [comment]
Raised https://github.com/lm-tools/choose-your-activities/pull/131 to added the review comments

**Pull request merged**